release/pkg/controller/operator/controller.go

```diff
@@ -0,0 +1,250 @@
+package operator
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/projectcalico/calico/release/internal/command"
+	"github.com/projectcalico/calico/release/internal/hashrelease"
+	"github.com/projectcalico/calico/release/internal/registry"
+	"github.com/projectcalico/calico/release/internal/utils"
+	"github.com/projectcalico/calico/release/pkg/controller/branch"
+)
+
+type OperatorController struct {
+	// Allow specification of command runner so it can be overridden in tests.
+	runner command.CommandRunner
+
+	// dockerRunner is for navigating docker
+	docker *registry.DockerRunner
+
+	// version is the operator version
+	version string
+
+	// repoRoot is the absolute path to the root directory of the repository
```

Comment:
Of which repository? I thought this was the repo root of the Calico repository (which I think is the default assumption unless otherwise stated), but it appears to be used as the root of the operator repo?

Might want to rename this to "operatorDir" to make that more clear.